odo/backends/sql.py

```diff
@@ -23,46 +24,88 @@
+    'real': sa.types.Float(precision=53),
+    'string': sa.types.Text,
+    'date': sa.types.Date,
+    'time': sa.types.Time,
+    'datetime': sa.types.DateTime,
+    'bool': sa.types.Boolean,
+    "timedelta[unit='D']": sa.types.Interval(day_precision=9,
+                                             second_precision=0),
+    "timedelta[unit='h']": sa.types.Interval(second_precision=0,
+                                             day_precision=0),
+    "timedelta[unit='m']": sa.types.Interval(second_precision=0,
+                                             day_precision=0),
+    "timedelta[unit='s']": sa.types.Interval(second_precision=0,
+                                             day_precision=0),
+    "timedelta[unit='ms']": sa.types.Interval(second_precision=3,
+                                              day_precision=0),
+    "timedelta[unit='us']": sa.types.Interval(second_precision=6,
+                                              day_precision=0),
+    "timedelta[unit='ns']": sa.types.Interval(second_precision=9,
+                                              day_precision=0),
+    # ??: sa.types.LargeBinary,
+    # Decimal: sa.types.Numeric,
+    # ??: sa.types.PickleType,
+    # unicode: sa.types.Unicode,
+    # unicode: sa.types.UnicodeText,
+    # str: sa.types.Text,  # ??
+}
 
 revtypes = dict(map(reversed, types.items()))
 
-revtypes.update({sa.types.VARCHAR: 'string',
-                 sa.types.String: 'string',
-                 sa.types.Unicode: 'string',
-                 sa.types.DATETIME: 'datetime',
-                 sa.types.TIMESTAMP: 'datetime',
-                 sa.types.FLOAT: 'float64',
-                 sa.types.DATE: 'date',
-                 sa.types.BIGINT: 'int64',
-                 sa.types.INTEGER: 'int',
-                 sa.types.NUMERIC: 'float64',  # TODO: extend datashape to decimal
-                 sa.types.BIGINT: 'int64',
-                 sa.types.NullType: 'string',
-                 sa.types.Float: 'float64'})
+revtypes.update({
+    sa.types.VARCHAR: 'string',
+    sa.types.String: 'string',
+    sa.types.Unicode: 'string',
+    sa.types.DATETIME: 'datetime',
+    sa.types.TIMESTAMP: 'datetime',
+    sa.types.FLOAT: 'float64',
+    sa.types.DATE: 'date',
+    sa.types.BIGINT: 'int64',
+    sa.types.INTEGER: 'int',
+    sa.types.NUMERIC: 'float64',  # TODO: extend datashape to decimal
+    sa.types.BIGINT: 'int64',
+    sa.types.NullType: 'string',
+    sa.types.Float: 'float64'
+})
+
+# interval types are special cased in discover_typeengine so remove them from
+# revtypes
+revtypes = valfilter(lambda x: not isinstance(x, sa.types.Interval), revtypes)
+
+
+units_of_power = {
+    0: 's',
+    3: 'ms',
+    6: 'us',
+    9: 'ns'
+}
 
 
 @discover.register(sa.sql.type_api.TypeEngine)
 def discover_typeengine(typ):
+    if isinstance(typ, sa.types.Interval):
+        if not (typ.second_precision or typ.day_precision):
+            return datashape.TimeDelta(units='us')
```

Comment:
I just pulled up the SQL2003 standard (I couldn't easily get access to 2008 or 2011) and there's only a default precision for non leading fields. For example if i specify a `day interval`, then by default the rest of the fields (`hour`, `minute`, and `second`) all have a precision of two integers (in the case of `second` this is two integers before the decimal point, since `second` intervals can be floating point values)